lib/mathlib.cpp

```diff
@@ -549,7 +549,7 @@ MathLib::bigint MathLib::toLongNumber(const std::string & str)
     biguint ret = 0;
     std::istringstream istr(str);
     istr >> ret;
-    return ret;
+    return static_cast<bigint>(ret);
```

Comment:
It seems unfortunate to change the sign like this. I guess `ret` should be a bigint and if there is overflow ... I'm not sure if an exception would be preferable or how we should handle it..

Reply:
This will be done implicitly anyways - I just doing it explicitly so UBSAN know it is actually intended.

Reply:
I rather fix the warning properly.